### DISCUSSION TITLE
got faulty monochrome image when convert svg into jpeg

### DISCUSSION THREAD

**Discussion by totake0224** [ImageMagick/ImageMagick#6048] [Help] [answered]
### ImageMagick version

6.9.12-77  ( I build and install this from source)

### Operating system, version and so on

ubuntu 20.04

### Description

I got monochrome image when I converted color svg file into jpeg.
I used command 
convert image.svg image.jpg
result image looks like binary filtered.
I got color image when I convert svg into png.(convert image.svg image.png)
I'm very perplexed.
please tell me why imagemagick convert my svg image at such a strange manner.

attached svg image contains outer png image that not attached.
but you can check converted image by attached png and jpg.

original svg
![6NGOD](https://user-images.githubusercontent.com/42058913/217795486-396cfb90-fd6a-4070-baad-9571f37922e6.svg)
converted jpg
![6NGOD](https://user-images.githubusercontent.com/42058913/217795490-49a288a2-b867-4fd5-9fd5-1ab80326c96a.jpg)
converted png (looks same when I opened original svg at inkscape)
![6NGOD](https://user-images.githubusercontent.com/42058913/217795494-a3635985-10d8-48d5-b4ed-78883fa302c2.png)

**Comment by fmw42** [comment]
When I use IM 6.9.12.76 with Inkscape as the SVG renderer, I get

`convert 217795486-396cfb90-fd6a-4070-baad-9571f37922e6.svg x.jpg`

![x](https://user-images.githubusercontent.com/13458218/217888009-bf519553-acdb-4b0d-a447-be36019c47f8.jpg)

You have an old version of IM. You might want to upgrade. But install Inkscape and try again.  There are 3 possible SVG renderers: IM internal MSVG/XML, RSVG delegate, Inkscape in order of better functionality.  You may be using the IM internal one.

Add -verbose to your command to find out which you are using.

**Reply by totake0224** [reply]
I think I'm using inkscape as a SVG renderer both converting png and jpg.
convert -verbose test.svg test.png
'inkscape' '/tmp/magick-hHZDQl-pXPeyw1p2caBvkj19JLbdT3MW' --export-filename='/tmp/magick-kehNoTFnGl_K82R7Uyb8uFoNahxawRHS.png' --export-dpi='96' --export-background='rgb(100%,100%,100%)' --export-background-opacity='1' > '/tmp/magick-lah4c5MnDzPP95PfiFjnYPRf8Zsk4vcT' 2>&1
/tmp/magick-kehNoTFnGl_K82R7Uyb8uFoNahxawRHS.png PNG 630x474 630x474+0+0 8-bit sRGB 288170B 0.010u 0:00.012
test.svg SVG 630x474 630x474+0+0 8-bit sRGB 288170B 0.000u 0:00.000
test.svg=>test.png SVG 630x474 630x474+0+0 8-bit sRGB 284566B 0.250u 0:00.229

convert -verbose test.svg test.jpg
'inkscape' '/tmp/magick-q9OwCwy0vLibwFRjgUDsIpzfOcm3lv3I' --export-filename='/tmp/magick-2cgqxTvTSzvRBUXeGfPdEaevVGg4Qy5K.png' --export-dpi='96' --export-background='rgb(100%,100%,100%)' --export-background-opacity='1' > '/tmp/magick-pTq7UZikQXjfShZaZf2Ug4emlqZA9jkL' 2>&1
/tmp/magick-2cgqxTvTSzvRBUXeGfPdEaevVGg4Qy5K.png PNG 630x474 630x474+0+0 8-bit sRGB 288170B 0.010u 0:00.014
test.svg SVG 630x474 630x474+0+0 8-bit sRGB 288170B 0.000u 0:00.000
test.svg=>test.jpg PNM 630x474 630x474+0+0 8-bit sRGB 895875B 0.000u 0:00.009

I attach SVG not contain outer image file.
original svg
![test](https://user-images.githubusercontent.com/42058913/217969831-79c23977-5221-4aef-a1b6-259e5c6fae4a.svg)

converted png
![test](https://user-images.githubusercontent.com/42058913/217969834-60c5f3f3-5c12-4b23-a5d2-cbe10ba87506.png)

converted jpg
![test](https://user-images.githubusercontent.com/42058913/217969835-59a1c910-81d1-40cc-88d0-e6b4ec302071.jpg)

**Comment by fmw42** [comment]
I get your same result, but if I add -trim +repage, it removes the border.

`convert 217969831-79c23977-5221-4aef-a1b6-259e5c6fae4a.svg -trim +repage x.png`

![x](https://user-images.githubusercontent.com/13458218/217971365-d68730b9-9954-4e15-81e3-0d94139752ad.png)

**Reply by totake0224** [reply]
Please check my converted jpg image. ( Somehow jpg image is not displayed inline)
I have any problem with converted png image. it's reference of expected result of converted jpg.
I think last row of my verbose means something.
> test.svg=>test.jpg PNM 630x474 630x474+0+0 8-bit sRGB 895875B 0.000u 0:00.009 
Do you know what is 'PNM' ?

**Reply by totake0224** [reply]
I found problem.
I will explain it at top tree
Thank you.

**Reply by fmw42** [reply]
PNM is a common format used for standard processing on Linux. IM is using it as a temporary format. Not to worry.

Did you get a jpg output? Or was it something else?

**Reply by fmw42** [reply]
Your delegates.xml file may have been modified to generate PNG output in place of JPG from Inkscape.  You can edit your delegates.xml file and set it back if your image was PNM in place of JPG.  When I do

```
convert -verbose 217969831-79c23977-5221-4aef-a1b6-259e5c6fae4a.svg x.jpg

identify x.jpg

x.jpg JPEG 630x474 630x474+0+0 8-bit sRGB 53441B 0.000u 0:00.000

```

I get a JPEG output.

**Reply by totake0224** [reply]
Thank you. Delegates is the problem.
I fixed this by rebuild imagemagick with certain jpeg library.

**Comment by totake0224** [comment, marked as answer]
I found problem.
I did'nt install libjpeg62-dev.
so there is no Delegate of jpeg.
after instal it. and build from source.
I could convert svg into jpg as I expected.
I refer to 
https://gist.github.com/nickferrando/fb0a44d707c8c3efd92dedd0f79d2911
to identify which library is connected to jpeg Delegate.

Thank you all. and sorry for bothering such a basic mistake.

**Comment by fmw42** [comment]
Note, next time post to the Imagemagick 6 GIT forum since you are using IM 6. This is an IM 7 forum.

**Reply by totake0224** [reply]
I'm sorry. I will
